packages/message-relayer/src/relay-tx.ts

```diff
@@ -0,0 +1,407 @@
+/* Imports: External */
+import { ethers } from 'ethers'
+import {
+  fromHexString,
+  remove0x,
+  toHexString,
+  toRpcHexString,
+} from '@eth-optimism/core-utils'
+import { getContractInterface, predeploys } from '@eth-optimism/contracts'
+import * as rlp from 'rlp'
+import { MerkleTree } from 'merkletreejs'
+
+// Number of blocks added to the L2 chain before the first L2 transaction. Genesis are added to the
+// chain to initialize the system. However, they create a discrepancy between the L2 block number
+// the index of the transaction that corresponds to that block number. For example, if there's 1
+// genesis block, then the transaction with an index of 0 corresponds to the block with index 1.
+const NUM_L2_GENESIS_BLOCKS = 1
```

Comment:
Will this supersede one of these config options? - https://github.com/ethereum-optimism/optimism/blob/77108d37c344a11bcdaca0013709b25c49036fb1/packages/message-relayer/src/exec/run.ts#L39-L50

Reply:
It'll probably supersede `L2_BLOCK_OFFSET`. I came to feel that this isn't something that should be user-configurable. I can't really see a case where we'll have more than one genesis block (and hence an offset of more than `1`). We either fix the offset problem entirely or we have the offset problem for now. But making it configurable doesn't make much sense to me.